lib/elixir/lib/enum.ex

```diff
@@ -717,6 +717,32 @@ defmodule Enum do
   """
   @spec fetch(t, integer) :: {:ok, element} | :error
   def fetch(enumerable, index)
+  def fetch(first.._, 0),
```

Comment:
I would like to do some benchmarks as I haven't done none yet. Let's see what's the performance like ;-D